# Analisa Data Tinggi Permukaan Air DKI Jakarta 2009 - 2014
Suisbiyanto Prasetya


Data berikut ini merupakan hasil pencatatan ketinggian permukaan Air pada pintu-pintu air di Jakarta tahun 2009-2014. Untuk Kondisi Aman permukaan Air yaitu antara -180 cm hingga 50 cm.

In [10]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

data_air = pd.read_csv('tinggi-permukaan-air-pintu-air.csv', delimiter = ',', encoding = "ISO-8859-1")
data_air.sample(20)
# data_air

nama_pintu_air     tanggal       jam  tinggi_air keterangan_cuaca  \
364664   Sunter Utara  2011-12-08  12:00:00        29.0                T   
8797       Angke Hulu  2013-01-03  20:00:00        40.0                T   
256045   Pesanggrahan  2011-03-02  14:00:00        65.0                T   
347368   Sunter Utara  2013-12-29  23:00:00        64.0                T   
263350    Pulo Gadung  2013-09-03  16:00:00       365.0                T   
139871      Katulampa  2011-03-28  23:00:00        10.0               MT   
21712      Angke Hulu  2011-07-08  21:00:00        40.0               MT   
268528    Pulo Gadung  2013-01-20  12:00:00       365.0                M   
146295    Krukut Hulu  2013-11-08  23:00:00        60.0               MT   
260808    Pulo Gadung  2013-12-23  19:00:00       365.0               MT   
297482    Sunter Hulu  2013-01-20  11:00:00        50.0                T   
56736   Cipinang Hulu  2010-05-26  17:00:00        85.0                T   
397761    Waduk Pluit  2011-06-04  09:00:00      -165.0                T   
297756    Sunter Hulu  2013-01-08  14:00:00        50.0                M   
403688    Waduk Pluit  2010-03-30  16:00:00      -115.0                G   
401590    Waduk Pluit  2010-12-17  13:00:00      -180.0                T   
234428   Pesanggrahan  2013-09-05  04:00:00        50.0                T   
41809   Cipinang Hulu  2012-07-11  14:00:00        75.0                T   
125064      Katulampa  2012-12-08  17:00:00       110.0                M   
317261    Sunter Hulu  2010-04-24  12:00:00        50.0                T   

         status  
364664   Normal  
8797     Normal  
256045   Normal  
347368   Normal  
263350   Normal  
139871   Normal  
21712    Normal  
268528   Normal  
146295   Normal  
260808   Normal  
297482   Normal  
56736    Normal  
397761   Normal  
297756   Normal  
403688   Normal  
401590   Normal  
234428   Normal  
41809    Normal  
125064  Waspada  
317261   Normal

###### T : Terang, MT : Mendung Tipis, M : Mendung, G : Derimis, H : Hujan


sumber data : https://data.go.id/dataset/laporan-tinggi-air-dki-jakarta

In [3]:
data_air.columns
# data_air[['nama_pintu_air','keterangan_cuaca','status']]

Index([u'nama_pintu_air', u'tanggal', u'jam', u'tinggi_air',
       u'keterangan_cuaca', u'status'],
      dtype='object')

In [4]:
# Tinggi Air Rata-rata di setiap pintu air
print "Tinggi Air 2009 - 2014 (cm):"
ax = data_air.groupby('nama_pintu_air').tinggi_air.agg(['mean', 'min', 'max'])
print (ax)

Tinggi Air 2009 - 2014 (cm):
                      mean     min     max
nama_pintu_air                            
Angke Hulu       61.964708  -175.0   550.0
Cipinang Hulu    81.985061     5.0   440.0
Depok           101.812807     8.0   730.0
Karet           402.272633  -180.0   760.0
Katulampa        25.961942     1.0   670.0
Krukut Hulu      62.097338  -180.0   765.0
Manggarai       684.467012  -160.0  7110.0
Pasar Ikan      164.146744  -190.0  2010.0
Pesanggrahan     72.235394    10.0   710.0
Pulo Gadung     369.583287    10.0   980.0
Sunter Hulu      53.091201     2.0   500.0
Sunter Selatan   91.494757   -66.0  8100.0
Sunter Utara     38.161580  -180.0  1001.0
Waduk Pluit    -151.954952 -1808.0   430.0


In [15]:
bx = data_air.groupby('nama_pintu_air').median()
print "Median Tinggi Air 2009 - 2014 (cm):"
print (bx)

Median Tinggi Air 2009 - 2014 (cm):
                tinggi_air
nama_pintu_air            
Angke Hulu            40.0
Cipinang Hulu         80.0
Depok                100.0
Karet                400.0
Katulampa             20.0
Krukut Hulu           60.0
Manggarai            680.0
Pasar Ikan           164.0
Pesanggrahan          70.0
Pulo Gadung          360.0
Sunter Hulu           50.0
Sunter Selatan        97.0
Sunter Utara          35.0
Waduk Pluit         -160.0


In [17]:
# cx = data_air[['nama_pintu_air', 'tinggi_air','status']]
# print (cx)

In [12]:
data_air.sort_values(by = "nama_pintu_air")
print "Ketinggian rata-rata, minimum, dan maksimum pada kondisi Kritis, Normal, Rawan, Waspada (cm):"
data_air.groupby(['nama_pintu_air', 'status']).tinggi_air.agg(['mean', 'min', 'max'])

Ketinggian rata-rata, minimum, dan maksimum pada kondisi Kritis, Normal, Rawan, Waspada (cm):


mean     min     max
nama_pintu_air status                              
Angke Hulu     Kritis    424.545455   375.0   550.0
               Normal     46.133714  -175.0    95.0
               Rawan     245.778824   200.0   345.0
               Waspada   134.169045   100.0   200.0
Cipinang Hulu  Kritis    413.333333   400.0   440.0
               Normal     81.848932     5.0   145.0
               Waspada   168.382353   150.0   195.0
Depok          Kritis    544.761905   360.0   730.0
               Normal    100.100024     8.0   195.0
               Rawan     291.714286   270.0   345.0
               Waspada   220.892193   200.0   265.0
Karet          Kritis    658.367347   610.0   760.0
               Normal    396.179398  -180.0   445.0
               Rawan     562.905983   550.0   600.0
               Waspada   472.309132    45.0   550.0
Katulampa      Kritis    451.666667   220.0   670.0
               Normal     24.231992     1.0    75.0
               Rawan     163.076923   150.0   180.0
               Waspada    92.986056    80.0   140.0
Krukut Hulu    Kritis    524.250000   390.0   765.0
               Normal     61.097954  -180.0   145.0
               Waspada   162.021645   150.0   195.0
Manggarai      Kritis   1859.750000   960.0  7110.0
               Normal    679.820623  -160.0   745.0
               Rawan     899.219388   850.0   945.0
               Waspada   776.095380   750.0   850.0
Pasar Ikan     Kritis    419.047619   231.0  2010.0
               Normal    146.847883  -190.0   169.0
               Rawan     199.408479   190.0   229.0
               Waspada   177.909756   170.0   189.0
Pesanggrahan   Kritis    577.272727   360.0   710.0
               Normal     70.746936    10.0   145.0
               Rawan     282.066667   250.0   320.0
               Waspada   177.563863   150.0   245.0
Pulo Gadung    Kritis    976.000000   970.0   980.0
               Normal    368.362329    10.0   545.0
               Rawan     721.875000   700.0   740.0
               Waspada   583.423358   550.0   685.0
Sunter Hulu    Kritis    386.666667   360.0   500.0
               Normal     52.414995     2.0   135.0
               Rawan     200.000000   200.0   200.0
               Waspada   154.803922   140.0   180.0
Sunter Selatan Kritis    418.377049   221.0  8100.0
               Normal     82.847391   -66.0    99.0
               Rawan     175.947059   150.0   218.0
               Waspada   104.653470   100.0   149.0
Sunter Utara   Kritis    204.239437   153.0  1001.0
               Normal     36.250559  -180.0    79.0
               Rawan     133.267857   120.0   148.0
               Waspada    85.779570    80.0   118.0
Waduk Pluit    Kritis     63.210526  -190.0   430.0
               Normal   -159.402724 -1808.0  -105.0
               Rawan     -39.405253   -60.0   -11.0
               Waspada   -86.974975  -100.0   -15.0

Dari tabel  tersebut terlihat standar status ketinggian air untuk masing-masing pintu air berbeda-beda sebagai contoh:
#### Waduk Pluit
- Normal antara -1808 sampai -105 cm
- Waspada antara -100 sampai -15 cm
- Rawan antara -60 sampai -11 cm
- Kritis antara -190 sampai 430 cm

#### Cipinang Hulu
- Normal antara 5 sampai 145 cm
- Waspada 150 sampai 195 cm
- Kritis antara 400 sampai 440 cm

Waduk pluit peringatan sudah diberikan pada ketinggian air > -100 cm, sedangkan untuk Cipinang Hulu peringatan baru diberikan jika ketinggian air > 150 cm.

In [29]:
ax = data_air.groupby('nama_pintu_air').mean()['tinggi_air']
ax = ax.plot(kind='bar', title ="Tinggi air Rata-Rata di setiap Pintu Air (2009 - 2014)", legend=True, fontsize=12)
ax.set_xlabel("Tanggal", fontsize=10)
ax.set_ylabel("Jumlah", fontsize=10)
plt.show()

<matplotlib.text.Text at 0xadfaf70>

Urutan Rata-rata tinggi air tertinggi yaitu:
- pintu AIR Manggarai
- Karet
- Pulo Gadung